### PR TITLE
Add build script for Context2 API

### PR DESCRIPTION
**Pull request by felixhammerl**
Allows building the library with Context2 API via:
```
$ ./do.sh build_library_ctx2
```

### CODE REVIEW

File: do.sh [hunk 1]
```diff
@@ -152,7 +152,7 @@ e2e_build_jsmodule() {
 }
 
 e2e_build_library() {
```
Comment (koto) [inline do.sh]: Can you rename the e2e_build_library to e2e_build_library_ctx1 and have the new e2e_build_library call that instead? I wanted to switch the defaults at some point.

File: do.sh [hunk 2]
```diff
@@ -165,6 +165,20 @@ e2e_build_library() {
...
 
+e2e_build_library_ctx2() {
+  echo "Building End-To-End library with Context2 API into $BUILD_DIR/library..."
+  e2e_assert_dependencies
+  set -e
+  e2e_assert_jsdeps
+  BUILD_EXT_DIR="$BUILD_DIR/library"
+  rm -rf "$BUILD_EXT_DIR"
+  mkdir -p "$BUILD_EXT_DIR"
+  e2e_build_closure_lib_ "e2e.openpgp.Context2Impl" "$BUILD_EXT_DIR/end-to-end.compiled.js"
```
Comment (koto) [inline do.sh]: Change the output filenames such that they don't overwrite the ctx1 ones.
We could then make a build target that builds both.

**Comment by felixhammerl** [comment]
alright, will do. need to check back with legal team for the cla.

**Comment by felixhammerl** [comment]
I have updated the build target to the following:

* `build_library` for Context2 API only
* `build_library_ctx1` for Context API only
* `build_library_all` for both, Context API suffixed with -ctx1

**Comment by sirdarckcat** [comment]
@koto did you take a look?

**Pull request merged**